theano/tensor/tests/test_subtensor.py

```diff
@@ -491,6 +491,8 @@ def test_ok_list(self):
                           # Test with TensorConstant index.
                           (rand(4, 2, 3),
                            theano.tensor.constant([3, 3, 1, 1, 2, 2, 0, 0])),
+                          # Test with non-contiguous input
+                          (rand(6, 2)[::2], [1, 0])
```

Comment:
This do not generate non-contiguous index. [1, 0] will be converted to a numpy array that will be contibuous.

I think you need to make a new test just for that.